mixer/adapter/redisquota/redisquota.go

```diff
@@ -0,0 +1,509 @@
+// Copyright 2017 Istio Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package redisquota provides a quota implementation with redis as backend.
+// The prerequisite is to have a redis server running.
+package redisquota
+
+import (
+	"context"
+	"fmt"
+	"hash/fnv"
+	"io"
+	"reflect"
+	"sort"
+	"strconv"
+	"time"
+
+	"github.com/go-redis/redis"
+	"istio.io/istio/mixer/adapter/redisquota/config"
+	"istio.io/istio/mixer/pkg/adapter"
+	"istio.io/istio/mixer/pkg/status"
+	"istio.io/istio/mixer/template/quota"
+)
+
+// LUA rate-limiting algorithm scripts
+var (
```

Comment:
I think it'd be nicer to put this code in a separate file.

Reply:
Separated scripts files